frontend/src/metabase/query_builder/actions/core/initializeQB.ts

```diff
@@ -247,17 +247,17 @@ async function handleQBInit(
     return;
   }
 
-  if (hasCard) {
-    await handleDashboardParameters(card, {
+  if (deserializedCard?.dashcardId) {
+    card = await propagateDashboardParameters({
+      card,
       deserializedCard,
```

Comment:
We were checking this inside the `handleDashboardParameters` before, but IMO for clarity, the function should only be called if deserializedCard is a dashcard. Otherwise, there's clearly no need to propagate dashboard parameters.
https://github.com/metabase/metabase/pull/25539/files#diff-74a80e24cd9cea4e9a716c02394f89df3331c13a1227fa4b9d7a76d81165928aL131